dgraph/dgraph.go

```diff
@@ -17,21 +18,43 @@ type Dgraph struct {
 	Conn   *grpc.ClientConn
 }
 
-//New creates a new instance of DGraph
-func New(addr string) (*Dgraph, error) {
+//New creates a new instance of DGraphc
+func New(addr string, apiKey string, username string, password string, namespace string) (*Dgraph, error) {
+
 	// Dial a gRPC connection. The address to dial to can be configured when
 	// setting up the dgraph cluster.
 	if addr == "" {
 		addr = "localhost:9080"
 	}
-	conn, err := grpc.Dial(addr, grpc.WithInsecure())
+
+	let conn *grpc.ClientConn
+	let err error
+
+	if (apiKey == "" || namespace == "") {
```

Comment:
why does it do that when namespace is empty? Is this for backwards compatibility?

Reply:
@Alex-IS Removed it, not needed.